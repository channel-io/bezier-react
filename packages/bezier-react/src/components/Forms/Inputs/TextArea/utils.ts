import { type SemanticNames } from '~/src/foundation'

interface CalculateBgColorProps {
  focused: boolean
  hasError: boolean
  readOnly: boolean
}

export function getTextAreaBgColorSemanticName({
  focused,
  hasError,
  readOnly,
}: CalculateBgColorProps): SemanticNames {
  if (readOnly) { return 'bg-grey-lighter' }
  if (focused || hasError) { return 'bg-white-normal' }
  return 'bg-grey-lightest'
}
