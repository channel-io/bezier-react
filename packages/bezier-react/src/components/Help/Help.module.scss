.Help {
  line-height: 0;

  &:not([data-state="closed"]) {
    > .Icon {
      color: var(--txt-black-darkest);
    }
  }
}
