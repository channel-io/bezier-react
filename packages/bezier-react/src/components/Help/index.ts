export { Help } from './Help'
export type { HelpProps } from './Help.types'
