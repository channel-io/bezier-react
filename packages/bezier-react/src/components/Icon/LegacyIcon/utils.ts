/* Internal Dependencies */
// eslint-disable-next-line no-restricted-imports
import icons, { type IconName } from '../generated'

/**
 * @deprecated Please import and use individual icons.
 * @example
 * import { Icon, AllIcon, type IconProps } from '@channel.io/bezier-react'
 * <Icon source={AllIcon} color="bg-black-dark" />
 * @example <caption>How to validate the bezier icon source</caption>
 * import { isBezierIcon, AllIcon } from '@channel.io/bezier-react'
 * isBezierIcon(AllIcon) // true
 * isBezierIcon(<svg>...</svg>) // false
 * @example <caption>Legacy icon component is still available. but it will be removed in future versions!</caption>
 * import { LegacyIcon, type LegacyIconProps } from '@channel.io/bezier-react'
 * <LegacyIcon name="all" color="bg-black-dark" />
 */
export const isIconName = (name: any): name is IconName => Object.keys(icons).includes(name)
