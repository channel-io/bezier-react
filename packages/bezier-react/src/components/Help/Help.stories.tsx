import type { Meta, StoryObj } from '@storybook/react'

import { Help } from './Help'

const meta: Meta<typeof Help> = {
  component: Help,
  argTypes: {
    children: {
      control: {
        type: 'text',
      },
    },
  },
}
export default meta

export const Primary: StoryObj<typeof Help> = {
  args: {
    children: 'Lorem ipsum',
  },
}
