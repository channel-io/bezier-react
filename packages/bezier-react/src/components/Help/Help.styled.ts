/* Internal dependencies */
import { styled } from '~/src/foundation'
import { Tooltip as BaseTooltip } from '~/src/components/Tooltip'
import { Icon as BaseIcon } from '~/src/components/Icon'

export const Icon = styled(BaseIcon)``

export const Tooltip = styled(BaseTooltip)`
  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    > ${Icon} {
      color: ${({ foundation }) => foundation?.theme?.['txt-black-darkest']};
    }
  }
`
