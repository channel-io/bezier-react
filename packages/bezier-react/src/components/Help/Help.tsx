import React, { forwardRef } from 'react'

import { HelpFilledIcon } from '@channel.io/bezier-icons'

import { isEmpty } from '~/src/utils/type'

import { IconSize } from '~/src/components/Icon'
import { Tooltip } from '~/src/components/Tooltip'

import type HelpProps from './Help.types'

import * as Styled from './Help.styled'

export const HELP_TEST_ID = 'bezier-react-help'
export const HELP_DISPLAY_NAME = 'Help'

const Help = forwardRef<HTMLDivElement, HelpProps>(function Help({
  children,
  ...rest
}, forwardedRef) {
  if (isEmpty(children)) { return null }

  return (
    <Tooltip
      {...rest}
      ref={forwardedRef}
      content={children}
    >
      <Styled.Trigger>
        <Styled.Icon
          testId={HELP_TEST_ID}
          source={HelpFilledIcon}
          size={IconSize.XS}
          color="txt-black-dark"
        />
      </Styled.Trigger>
    </Tooltip>
  )
})

Help.displayName = HELP_DISPLAY_NAME

export default Help
