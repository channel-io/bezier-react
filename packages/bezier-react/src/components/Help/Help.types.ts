import {
  type AlphaBezierComponentProps,
  type ChildrenProps,
} from '~/src/types/ComponentProps'

import { type TooltipProps } from '~/src/components/Tooltip'

export default interface HelpProps extends
  AlphaBezierComponentProps,
  ChildrenProps,
  Omit<TooltipProps, 'content' | 'children'> {}
