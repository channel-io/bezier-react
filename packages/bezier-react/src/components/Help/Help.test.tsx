import React from 'react'

import { render } from '~/src/utils/testUtils'

import Help, { HELP_TEST_ID } from './Help'
import type HelpProps from './Help.types'

describe('Help >', () => {
  let props: HelpProps

  beforeEach(() => {
    props = {
      children: 'Lorem ipsum',
    }
  })

  const renderComponent = (otherProps?: Partial<HelpProps>) =>
    render(<Help {...props} {...otherProps} />)

  it('Snapshot >', () => {
    const { getByTestId } = renderComponent()
    const rendered = getByTestId(HELP_TEST_ID)

    expect(rendered).toMatchSnapshot()
  })

  it('renders text when children prop is not empty', () => {
    const { getByTestId } = renderComponent()
    const rendered = getByTestId(HELP_TEST_ID)

    expect(rendered).toBeInTheDocument()
  })

  it('renders nothing when children prop is empty', () => {
    const { queryByTestId } = renderComponent({ children: '' })
    const rendered = queryByTestId(HELP_TEST_ID)

    expect(rendered).toBeNull()
  })
})
