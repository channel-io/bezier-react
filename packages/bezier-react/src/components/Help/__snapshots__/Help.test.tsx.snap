// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help > Snapshot > 1`] = `
.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin: 0px 0px 0px 0px;
  color: #00000066;
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color;
  transition-property: color;
}

<svg
  class="c0 "
  color="txt-black-dark"
  data-testid="bezier-react-help"
  fill="none"
  foundation="[object Object]"
  height="16"
  marginbottom="0"
  marginleft="0"
  marginright="0"
  margintop="0"
  viewBox="0 0 24 24"
  width="16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clip-rule="evenodd"
    d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2Zm.83 12.541h-1.9v-.708c0-1.156.625-2.24 1.672-2.9.867-.546 1.337-.939 1.337-1.447 0-1.014-1-1.561-1.94-1.561-1.035 0-1.942.73-1.942 1.56h-1.9c0-1.875 1.758-3.46 3.841-3.46 2.154 0 3.842 1.52 3.842 3.46 0 1.655-1.393 2.533-2.225 3.056-.236.15-.784.573-.784 1.292v.708Zm-.889 3.631a1.244 1.244 0 1 1 0-2.487 1.244 1.244 0 0 1 0 2.487Z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;
